apps/onlyoffice.sh

```diff
@@ -21,31 +18,58 @@ debug_mode
 root_check
 
 # Nextcloud 13 is required.
-lowest_compatible_nc 13
+lowest_compatible_nc 18
 
 # Test RAM size (2GB min) + CPUs (min 2)
 ram_check 2 OnlyOffice
 cpu_check 2 OnlyOffice
 
-# Notification
-msg_box "Before you start, please make sure that port 80+443 is directly forwarded to this machine!"
-
 # Get the latest packages
 apt update -q4 & spinner_loading
 
-# Check if Nextcloud is installed
-print_text_in_color "$ICyan" "Checking if Nextcloud is installed..."
-if ! curl -s https://"${NCDOMAIN//\\/}"/status.php | grep -q 'installed":true'
+# Check if onlyoffice is already installed
+print_text_in_color "$ICyan" "Checking if Onlyoffice is already installed..."
+if version_gt "$CURRENTVERSION" "18.0.1" && ! does_this_docker_exist 'onlyoffice/documentserver'
+then
+    install_if_not jq
+    if occ_command_no_check app:list --output=json | jq -e '.enabled | .documentserver_community' > /dev/null
+    then
+        choice=$(whiptail --radiolist "It seems like 'Onlyoffice' is already installed.\nChoose what you want to do.\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+        "Uninstall Onlyoffice" "" OFF \
+        "Reinstall Onlyoffice" "" ON 3>&1 1>&2 2>&3)
+
+        case "$choice" in
+            "Uninstall Onlyoffice")
+                print_text_in_color "$ICyan" "Uninstalling Onlyoffice..."
+                occ_command_no_check app:remove onlyoffice
+                occ_command app:remove documentserver_community
+		docker_prune_this 'onlyoffice/documentserver'
+                msg_box "Onlyoffice was successfully uninstalled."
+                exit
+            ;;
+            "Reinstall Onlyoffice")
+                print_text_in_color "$ICyan" "Reinstalling Onlyoffice..."
+                occ_command_no_check app:remove onlyoffice
+                occ_command app:remove documentserver_community
+            ;;
+            *)
+            ;;
+        esac
+	fi
+else
+        print_text_in_color "$ICyan" "Installing OnlyOffice..."
+fi
+
+# Check if Nextcloud is installed with SSL
+if ! occ_command_no_check config:system:get overwrite.cli.url | grep -q "https"
 then
-msg_box "It seems like Nextcloud is not installed or that you don't use https on:
-${NCDOMAIN//\\/}.
-Please install Nextcloud and make sure your domain is reachable, or activate SSL
-on your domain to be able to run this script.
-If you use the Nextcloud VM you can use the Let's Encrypt script to get SSL and activate your Nextcloud domain.
-When SSL is activated, run these commands from your terminal:
-sudo curl -sLO $APP/onlyoffice.sh
-sudo bash onlyoffice.sh"
-    exit 1
+msg_box "Sorry, but Nextcloud needs to be run on HTTPS which doesn't seem to be the case here.
+
+You easily activate TLS (HTTPS) by running the Let's Encrypt script found in $SCRIPTS.
+More info here: https://bit.ly/37wRCin
```

Comment:
We don't need https here, to make it work, since it also works without https.
What I wanted to do is to inform the user, that he can reinstall it later on, when ssl is activated to make it work again after https is activated.

Reply:
no, that is not right. is better if we just allow https.